include/libtorrent/utp_socket_manager.hpp

```diff
@@ -130,23 +130,28 @@ namespace libtorrent
 		typedef std::multimap<std::uint16_t, utp_socket_impl*> socket_map_t;
 		socket_map_t m_utp_sockets;
 
+		typedef std::vector<utp_socket_impl*> socket_vector_t;
+
 		// this is a list of sockets that needs to send an ack.
 		// once the UDP socket is drained, all of these will
 		// have a chance to do that. This is to avoid sending
 		// an ack for every single packet
-		std::vector<utp_socket_impl*> m_deferred_acks;
+		socket_vector_t m_deferred_acks;
+		socket_vector_t m_deferred_acks_tock;
```

Comment:
in fact, all of the ``_tock`` vectors could be collapsed to a single one, called something like ``temp``. The only purpose of it is to make it possible for events happening while iterating over it to add new sockets to the main vector. In fact, that temporary vector could be ``static``, since uTP sockets are single threaded, only one at a time will be using that temporary vector.